From 1df93f9c6fa4974581dfdadd948ac72568b5c99f Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Mon, 08 Jul 2024 17:58:24 +0300
Subject: [PATCH] [FSW-420] fs1: copy manifest of requested image

with this patch, added api support to copy
manifest for the requested image type and
return arc dma destination address.

Change-Id: I11abf55ab92b575a1f409eb721731952ed802363
---

diff --git a/common/boot_desc.c b/common/boot_desc.c
index 3046937..3fce0b1 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -228,3 +228,18 @@
 
 	return BOOT_ROM_IMAGE_NOT_FOUND;
 }
+
+uint32_t boot_desc_param_get(enum hl_fw_boot_ptr_imgs img, struct hl_fw_boot_ptrs *boot_ptr)
+{
+	boot_desc_update_ptrs(img);
+
+	if (!p_boot_ptr)
+		return 0;
+
+	memcpy(boot_ptr, p_boot_ptr, sizeof(struct hl_fw_boot_ptrs));
+
+	if (img == HL_FW_BOOT_PTR_RECOVERY)
+		return BOOT_PTR_REC_DMA_DEST_ARC;
+	else
+		return BOOT_PTR_SPI_DMA_DEST_ARC;
+}
diff --git a/common/boot_desc.h b/common/boot_desc.h
index 1a8b10e..1028a05 100644
--- a/common/boot_desc.h
+++ b/common/boot_desc.h
@@ -28,6 +28,7 @@
 int boot_desc_init(enum hl_fw_boot_ptr_imgs img);
 uint32_t boot_desc_get_img_desc(enum hl_fw_boot_ptr_imgs img, int type,
 				struct boot_desc *desc, enum fw_post_reset_cmd upd_status);
+uint32_t boot_desc_param_get(enum hl_fw_boot_ptr_imgs img, struct hl_fw_boot_ptrs *boot_ptr);
 
 #endif /* _BOOT_DESC_H_ */
 
